google/cloud/spanner_dbapi/cursor.py

```diff
@@ -394,38 +398,48 @@ def setoutputsize(self, size, column=None):
         """A no-op, raising an error if the cursor or connection is closed."""
         self._raise_if_closed()
 
+    def _handle_DQL_with_snapshot(self, snapshot, sql, params):
+        # Reference
+        #  https://googleapis.dev/python/spanner/latest/session-api.html#google.cloud.spanner_v1.session.Session.execute_sql
+        sql, params = parse_utils.sql_pyformat_args_to_spanner(sql, params)
+        res = snapshot.execute_sql(
+            sql, params=params, param_types=get_param_types(params)
+        )
+        if type(res) == int:
```

Comment:
Doesn't this mean that the statement was an update statement? In that case, it should not be possible to ever enter this if-block, as an update statement cannot be executed on a read-only snapshot.
Or is this some specific handling for `SELECT COUNT(*)` statements?

Reply:
I don't think it's a relevant part of code at all, it was probably copy-pasted from somewhere else. `Snapshot.execute_sql()` returns `StreamedResultSet`, and at the first look I don't see how an `int` can get here.

Still, in this PR I wasn't going to change anything, I've only rewritten the method to make it able to pass a snapshot as an arg (not only use it as a context). So, there is a note for this `if` statement, but let's keep it for another PR.